app/assets/javascripts/posts/editor.js

```diff
@@ -406,23 +412,28 @@ function setGalleriesAndDefault(galleries, defaultIcon) {
   $("#current-icon").addClass('pointer');
 
   // Calculate new galleries
-  var multiGallery = galleries.length > 1;
-  for (var j = 0; j < galleries.length; j++) {
-    iconSelectBox.append(galleryString(galleries[j], multiGallery));
-  }
+  setGalleries(galleries);
+
+  if (defaultIcon && shownIcons.indexOf(defaultIcon.id) < 0) { iconSelectBox.append(iconString(defaultIcon)); }
 
-  // If no default and no icons in any galleries, remove pointer
-  if (!defaultIcon && shownIcons.length === 0) {
+  // If no icons, remove pointer
+  if (shownIcons.length === 0) {
```

Comment:
Not clear on this change 🤔 If there's a default icon and no gallery, won't this now fail to trigger?

Reply:
It wouldn't trigger with a default icon before, either, though

Reply:
The inverse, rather, there's _no_ default icon and no gallery, my bad

Reply:
If there's no default icon the length should be 0? I think?

Reply:
Maaaybe? shownIcons is a global and so I'm not sure where all we touch it from =( this might be fine but I don't know for sure

Reply:
Actually, why do we even remove the pointer here if there _is_ a default icon? This code is weird – I'd probably leave it until we can be confident in testing it

Reply:
Reverted pending What Is This Code Even